docs/deployment-ux.md

```diff
@@ -300,111 +310,193 @@ impacts
-    SomeEnvironment/Prod:2    Deploying    5         1         1            1        
+    ```
+    $ aws ecs describe-environment-status --all --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT
+    SomeEnvironment/Prod:1    Undeploying  0         4
+    SomeEnvironment/Prod:2    Deploying    5         1
     ```
 
     Deployment progresses in a similar fashion until the last task of
     the old revision is left:
 
-    ``` {.shell}
-    $ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
-    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
-    SomeEnvironment/Prod:1    Undeploying  0         1         1            1        
-    SomeEnvironment/Prod:2    Deploying    5         5         5            5        
+    ```
+    $ aws ecs describe-environment-status --all --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT
+    SomeEnvironment/Prod:1    Undeploying  0         1
+    SomeEnvironment/Prod:2    Deploying    5         5
     ```
 
     Deployment completes once the final task in the old revision is
     terminated:
 
-    ``` {.shell}
-    $ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
-    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
-    SomeEnvironment/Prod:1    Inactive     0         0         0            0        
-    SomeEnvironment/Prod:2    Active       5         5         5            5        
     ```
+    $ aws ecs describe-environment-status --all --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT
+    SomeEnvironment/Prod:1    Inactive     0         0
+    SomeEnvironment/Prod:2    Active       5         5
+    ```
+
+#### Changes to cluster membership while a deployment is ongoing
+
+For larger clusters, or clusters that are backed by Autoscaling Groups, the available instances in the cluster might change (sometimes dramatically). Since the total number of available instances determines the desired end-state of the Environment, these state changes could affect the ability of an Environment to make progress while honoring its deployment constraints.
+
+The behaviour we're aiming for is to minimize the number of instances that are not running the Daemon task at all at any given point in time. If a configurable number of instances are without the correct version of a Daemon task for a configurable amount of time, the deployment will enter the `Failing` state and emit an alarm. However, it will continue to try and make progress.
+
+- If a deployment fails to start a task on a particular host (e.g. because the task definition is faulty, or the Task fails to start up in a nondeterministic way), priority will be given to restarting that task, rather than continuing to replace existing tasks.
```

Comment:
I think we should have a start/update deployment section and this should go there.

Reply:
> start/update deployment section

Could you clarify? What kind of detail would you like to see in that section? I don't think I understand how this is different from the general description of first launch at the start of the doc.

Reply:
As discussed in person, will extract the invariant here into a section, and then discuss the implications for the state changed/no state changed scenarios.